packages/editor-ui/src/components/RunData.vue

```diff
@@ -1199,47 +1167,59 @@ export default defineComponent({
 			const itemsLabel = itemsCount > 0 ? ` (${itemsCount} ${items})` : '';
 			return option + this.$locale.baseText('ndv.output.of') + (this.maxRunIndex + 1) + itemsLabel;
 		},
-		getDataCount(
+		getData(
 			runIndex: number,
 			outputIndex: number,
 			connectionType: ConnectionTypes = NodeConnectionType.Main,
-		) {
-			if (this.pinData) {
-				return this.pinData.length;
-			}
+		): INodeExecutionData[] {
+			let inputData: INodeExecutionData[] = [];
 
-			if (this.node === null) {
-				return 0;
+			if (this.node) {
+				inputData = this.getNodeInputData(
+					this.node,
+					runIndex,
+					outputIndex,
+					this.paneType,
+					connectionType,
+				);
 			}
 
-			const runData: IRunData | null = this.workflowRunData;
-
-			if (runData === null || !runData.hasOwnProperty(this.node.name)) {
-				return 0;
+			if (inputData.length === 0 || !Array.isArray(inputData)) {
+				return [];
 			}
 
-			if (runData[this.node.name].length <= runIndex) {
+			return inputData;
+		},
+		getPinDataOrInputData(inputData: INodeExecutionData[]): INodeExecutionData[] {
```

Comment:
Suggestion - Pin data is also "input" data, so it can get confusing:

```suggestion
		getPinDataOrLiveData(inputData: INodeExecutionData[]): INodeExecutionData[] {
```

Reply:
done